src/pages/shop/[id]/[name]/index.tsx

```diff
@@ -0,0 +1,120 @@
+import Breadcrumbs from "components/Breadcrumbs";
+import Layout from "components/Layout/Layout";
+import MaxWidth from "components/MaxWidth";
+import { rootDomain } from "lib/constants";
+import { identity } from "lodash";
+import Shop from "models/Shop";
+import React, { useEffect } from "react";
+import styles from "../../../Beach/[[...slug]]/styles.module.css";
+import ShopPageHero from "components/ShopPage/ShopPageHero";
+import ShopDetails from "components/ShopPage/ShopDetails";
+import Carousel from "components/Carousel/Carousel";
+import Error from 'next/error';
+interface Context {
+  params: {
+    id: string;
+    name: string;
+  };
+}
+
+export async function getStaticProps(context: Context) {
+  let shopData: any = fetch(`${rootDomain}/shop/get/${context.params.id}`);
+
+  shopData = await shopData.then((res: any) => {
+    if (res.status == 404) {
+      return {
+          errorCode: 404,
+      }
+    }
+    return res.json();
+  });
+
+  if (!shopData || (shopData.data && shopData.data.errorCode)) {
+    return {
+      notFound: true,
+    }
+  }
+
+  return {
+    props: {
+      'shop': shopData.data,
+    }, // will be passed to the page component as props
+    revalidate: 3600,
+  };
+}
+
+export async function getStaticPaths() {
+  const res = await fetch(`${rootDomain}/shop/get`);
+  const data = await res.json();
+
+  return {
+    paths: data.data.map((shop: Shop) => ({
+      params: {
+        id: `${shop.id}`,
+        name: `${shop.name ? shop.name : null}`,
+      },
+    })),
+    fallback: "blocking",
+  };
+}
+
+function ShopPage(props: any) {
+  const [shop, setShop] = React.useState(props.shop);
+  const [nearbyBeaches, setNearbyBeaches] = React.useState([])
+
+  useEffect(() => {
+    let nearby = fetch(`${rootDomain}/spots/nearby?lat=${props.shop.latitude}&lng=${props.shop.longitude}`)
+    nearby.then((res) => {
+      return res.json()
+    }).then(data => {
+      setNearbyBeaches(data.data)
+    })
+  }, [shop])
```

Comment:
it looks like there's a compiler warning and it wants you to change this to `[spots.latitude, spots.longitude]`